File: Cryptography and security libraries and algorithms/Javascript/Replit Ghostwriter/Timing attacks.js
const crypto = require('crypto');
const password = Buffer.from('password');
const userInput = Buffer.from('user-input');
//Log the user in if the password and user input match
if (crypto.createHash('sha256').update(password).digest('hex') === crypto.createHash('sha256').update(userInput).digest('hex')) {
  console.log('User is logged in');
} else {
  console.log('User is not logged in');
}

File: Cryptography and security libraries and algorithms/Python/Code complete/Timing attacks.py
import hmac

password = b'password'
user_input = b'user-input'

# Log the user in if the password and user input match
if hmac.compare_digest(hmac.new(password).hexdigest(),
                       hmac.new(user_input).hexdigest()):
    print('Logged in')
